### ISSUE TITLE
erros when db_bench linking with gflags

### ISSUE THREAD

**Issue by yncxcw** [facebook/rocksdb#1806]
hi, all
I am new to rocksdb, It woks fine when only build for rocksdb, but have erros when buld for db_bench.
I execute:
`make db_bench`
and the output error is:
 ```
CCLD     db_bench
tools/db_bench_tool.o: In function `__static_initialization_and_destruction_0(int, int) [clone .constprop.760]':
/home/cwei/project/rocksdb/tools/db_bench_tool.cc:80: undefined reference to `google::FlagRegisterer::FlagRegisterer<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
/home/cwei/project/rocksdb/tools/db_bench_tool.cc:183: undefined reference to `google::FlagRegisterer::FlagRegisterer<long>(char const*, char const*, char const*, long*, long*)'
/home/cwei/project/rocksdb/tools/db_bench_tool.cc:185: undefined reference to `google::FlagRegisterer::FlagRegisterer<long>(char const*, char const*, char const*, long*, long*)'
....
collect2: error: ld returned 1 exit status
Makefile:892: recipe for target 'db_bench' failed
make: *** [db_bench] Error 1
```

And I have installed gflags:
```
ls /usr/local/lib/libgflags*
/usr/local/lib/libgflags.a  /usr/local/lib/libgflags_nothreads.a
```

I am confusing why I still have the linking errors.

**Comment by wrinkledcheese** [comment]
I've duplicated this issue in #1820

**Comment by lgalanis** [comment]
I ran into the same issue. Here is what I figured out (Centos 7 gcc 4.8.5): 

1. Our build system will look for libraries only in the paths listed by g++ -print-search-dirs. Typically /usr/local/lib is not part of the list. The preprocessor will look in /usr/local/include, which leads to these confusions.
2. make sure the library and the build system agree on the namespace. nm -C /usr/local/lib/libgflags.a should agree with the output of build_tools/build_detect_platform
